plugins/modules/gitlab_project_members.py

```diff
@@ -89,6 +89,26 @@
     elements: str
     choices: ['guest', 'reporter', 'developer', 'maintainer']
     version_added: 3.7.0
+  gitlab_group:
+    description:
+      - A list of id or the full path of the group to share/unshare the project with.
+      - Every dictionary in this list specifies a group (by id or full path) and the access level the group should have.
+      - Mutually exclusive with O(gitlab_user), O(access_level) and O(gitlab_users_access).
+    type: list
+    elements: dict
+    suboptions:
+      name:
```

Comment:
I would use two different fields for ID and name, instead of having to figure out whether a string is an ID or a name.

Reply:
All the other modules that use the group variable work with both ID and name. I would leave it as it is for consistency

Reply:
@Lunik @Shaps @lgatellier @marwatk @metanovii @nejch @scodeman @sh0shin @suukit @waheedi @zanssa can someone of you take a look at this?